src/meshlink.c

```diff
@@ -2327,7 +2385,47 @@ static ssize_t channel_send(struct utcp *utcp, const void *data, size_t len) {
         free(hex);
     }
 
-    return meshlink_send(mesh, (meshlink_node_t *)n, data, len) ? len : -1;
+    // Prepare the packet
+    vpn_packet_t *packet = malloc(sizeof *packet);
+    if(!packet) {
+        meshlink_errno = MESHLINK_ENOMEM;
+        logger(mesh, MESHLINK_ERROR, "Error: channel_send packet memory allocation failed");
+        return false;
+    }
+
+    packet->probe = false;
```

Comment:
This is now duplicated code with `meshlink_send`. Can it be refactored into a common function that prepares the packet?